src/App.jsx

```diff
@@ -6,19 +6,25 @@ import AddTodoForm from './components/AddTodoForm';
 
 const Heading = () => <h1>Todo list</h1>
 
-const useSemiPersistentState = (key, initialValue) => {
-  const [value, setValue] = useState(
-    JSON.parse(localStorage.getItem(key)) || initialValue
-  );
+const App = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  useEffect(() => {
+    new Promise((resolve, reject) => 
+      setTimeout(
+        () => resolve({data: { todoList: JSON.parse(localStorage.getItem("savedTodoList")) } }),
```

Comment:
I think this also needs to handle the case of `savedTodoList` being null. right now, if this happens `todoList` will be set to `null` and the application will break since the `TodoList` component expects the prop to be an array type

Reply:
added "|| []"
would it fix it?